trino/client.py

```diff
@@ -197,14 +197,19 @@ def __setstate__(self, state):
         self._object_lock = threading.Lock()
 
     @property
-    def role(self):
+    def roles(self):
         with self._object_lock:
-            return self._role
+            return self._roles
 
-    @role.setter
-    def role(self, role):
+    @roles.setter
+    def roles(self, roles):
         with self._object_lock:
-            self._role = role
+            self._roles = roles
+
+    def _create_roles(self, roles: Optional[Dict[str, str]]) -> Optional[str]:
+        if roles is not None:
+            return ";".join(map(lambda role: f"{role[0]}={role[1]}", roles.items()))
```

Comment:
I'm not sure whether other clients do that maybe it's worth to catch incorrect roles value and put an error message about a proper roles syntax.

Reply:
Our param is a `Dict[str, str]`, we don't really check if the passed string is properly escaped if you supply a role containing chars like `=` or `;`. I would think it needs to follow quoted identifiers rules but not sure. I will check the JDBC client and see if I can find something.

Reply:
It seems to not work as expected: See https://github.com/trinodb/trino/issues/14252

Reply:
I don't think it should be handled by us, but be based on user input. I think the user should just pass in quoted input and it's up to the engine to correctly handle quoted identifiers:

Something like `{"hive":'"my=weird;role_name="'}`

Reply:
I agree, quoting should be left to user to perform.
We don't do it for them in queries or catalog/schema names - we shouldn't do it other places either.